app/actions/views/command.ts

```diff
@@ -39,15 +41,41 @@ export function executeCommand(message: string, channelId: string, rootId: strin
 
         const appsAreEnabled = appsEnabled(state);
         if (appsAreEnabled) {
-            const parser = new AppCommandParser({dispatch, getState}, args.channel_id, args.root_id);
+            const parser = new AppCommandParser({dispatch, getState}, intl, args.channel_id, args.root_id);
             if (parser.isAppCommand(msg)) {
                 const call = await parser.composeCallFromCommand(message);
+                const createErrorMessage = (errMessage: string) => {
+                    return {error: {message: errMessage}};
+                };
+
                 if (!call) {
                     return {error: {message: 'Error submitting command'}};
                 }
 
-                call.type = AppCallTypes.SUBMIT;
-                return dispatch(doAppCall(call, intl));
+                const res = await dispatch(doAppCall(call, AppCallTypes.SUBMIT, intl)) as {data: AppCallResponse};
+                const callResp = res.data;
+                switch (callResp.type) {
+                case AppCallResponseTypes.OK:
+                    if (callResp.markdown) {
+                        sendEphemeralPost(callResp.markdown, args.channel_id, args.parent_id);
+                    }
+                    return {data: {}};
```

Comment:
why is this returning an empty object in `data`? would it b sufficient to return `{data: true}` ?

Reply:
The expected data to be returned from here is an Slash Command Response (https://developers.mattermost.com/integrate/slash-commands/). Right now is not being used, so {} and true would mean almost the same, but I preferred to keep the empty object.

Reply:
As long as the empty object is being handled correctly I have no problem with it